lib/client/interfaces.go

```diff
@@ -384,14 +386,80 @@ func (k *Key) CertRoles() ([]string, error) {
 	return roles, nil
 }
 
+const (
+	agentCommentPrefix    = "teleport"
+	agentCommentSeparator = ":"
+)
+
+// teleportAgentKeyName returns a teleport agent key name like "teleport:<proxyHost>:<userName>".
+// This name should be used/seen in the comment section of Teleport agent keys, as the prefix for
+// Teleport agent cert keys.
+func teleportAgentKeyName(k KeyIndex) string {
+	return strings.Join([]string{agentCommentPrefix, k.ProxyHost, k.Username}, agentCommentSeparator)
+}
+
+// teleportAgentKeyCertName returns a teleport agent key cert name like "teleport:<proxyHost>:<userName>:<clusterName>".
```

Comment:
Nit: This may be my OCD speaking but can we make it `teleport:proxy:cluster:user` instead? :) To keep proper "hierarchy".